torch/csrc/jit/script/compiler.cpp

```diff
@@ -806,6 +806,60 @@ std::shared_ptr<SugaredValue> BuiltinFunction::call(
       loc, *m.graph(), symbol, self, inputs, attributes, true));
 }
 
+std::shared_ptr<SugaredValue> ForkValue::call(
+    SourceRange loc,
+    Method& m,
+    const std::shared_ptr<SugaredValue> &forked,
+    NamedValue& input) {
+  auto inputs = toValues(*m.graph(), {input});
+  if (inputs.size() != 1 || inputs.at(0)->type()->kind() != TypeKind::TupleType) {
+    throw ErrorReport(loc) << "Expected a single tuple argument to fork()";
+  }
+  auto n = m.graph()->insertNode(m.graph()->create(prim::Fork, inputs, 1))
+              ->setSourceLocation(std::make_shared<SourceRange>(loc));
+  auto body_block = n->addBlock();
+
+  std::vector<NamedValue> block_inputs;
+  auto schema = forked->schema(loc, m);
+  for (auto &arg : schema.arguments) {
+    Value *v = body_block->addInput()->setType(arg.type);
+    block_inputs.emplace_back(v);
+  }
+  Value *node_output;
+  {
+    WithInsertPoint guard(body_block);
+    // TODO: this is an issue
+    auto fn_sugared_output = forked->call(loc, m, block_inputs, {}, 1);
+    auto fn_simple_output = fn_sugared_output->asValue(loc, m);
+    body_block->registerOutput(fn_simple_output);
+    node_output = n->output()->setType(FutureType::create(fn_simple_output->type()));
+  }
+  return std::make_shared<SimpleValue>(node_output);
+}
+
+std::shared_ptr<SugaredValue> WaitValue::call(
```

Comment:
We do not need a WaitValue class, `wait` can be handled by the normal builtin operator pathway.

Simply register it as a known builtin (e.g. see jit/__init__.py):
```
_register_builtin(len, 'aten::len')
```